app/models/edition/timestamps.rb

```diff
@@ -0,0 +1,72 @@
+class Edition::Timestamps
+  def self.edited(
+    edition,
+    payload,
+    previous_live_version = nil,
+    now = Time.zone.now
+  )
+
+    if previous_live_version
+      edition.temporary_first_published_at = previous_live_version.first_published_at
+    end
+    edition.temporary_last_edited_at = now
+
+    edition.publisher_first_published_at = payload[:first_published_at]
+
+    unless edition.update_type == "major"
+      edition.major_published_at = previous_live_version&.major_published_at
+    end
+
+    if payload[:last_edited_at].present?
```

Comment:
Is it possible to do `edition.last_edited_at = payload[:last_edited_at] || now`?